### ISSUE TITLE
Speedez - User configurable permission access per each report

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1319]
In GitLab by @kdjstudios on Jan 10, 2019, 12:04

**Submitted by:** Jo Ann Browne <joann@speedez.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2019-01-10-86414

**Server:** External

**Client/Site:** Speed Ez

**Account:** NA

**Issue:** 

Lauren wants to know if we can choose what reports each person can be open to - not just an all or none?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 10, 2019, 12:15

@smasih @amishra @pchaudhary @tim.traylor 

After Jo Ann and their team was able to review the #1204. they have an additional request that we will need to estimate. 

I wrote:

> Thank you, I am going to open this in a separate ticket thread to track its estimate and progress independently, as it does differ significantly from the original request. I do have two initial questions to ask to insure the support team provides an accurate estimate. 
> 
> 1. Do you want an user interface made available to your login only to allow for you to configure which reports each member has access too? Or do you still want this to be something that the support team configures.
> 2. For each report, you simply want to be able to choose if an user has access to it, or is not able to access it. There is no other filtering or limitations needed?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 10, 2019, 13:44

Jo Ann wrote:

> Kyle that was it. Please look into whether this is possible or not. What I am trying to achieve is employee’s NOT have the rights to run ANY report that might tell them how much my company runs on a 30-69-90 day report. One thing they do need to know is what accounts are behind in payments etc.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 10, 2019, 13:55

Jo Ann Wrote:

> I would like to configure it by user.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 16, 2019, 13:15

@smasih @amishra @pchaudhary @tim.traylor 

May we please get an estimate on creating a user configurable UI to allow to select which reports a user has access too.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jan 22, 2019, 01:37

@kdjstudios @tim.traylor 

Sure, but first can we have the whole process laid out along with the mockups required. This will help in estimation, faster development and set expectations.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 22, 2019, 14:44

@smasih 

Thanks for requesting clarification, rather then just providing an inaccurate estimate. I will review this with @tim.traylor and create some Mock Ups, then get back to the client and approve those mock ups prior to having your team create an estimate.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 13, 2019, 14:57

Some initial mocks up to review with @tim.traylor 

The user button will be located at the top with the settings and logout. This will navigate to the user list below.

![Navigation_Button](/uploads/cec83eeab86b62e96665dfc6fa6279d3/Navigation_Button.png)

The user list will only display "non admin" users that have access to the same site as this manager user.

![User_List](/uploads/7092c1dbfdf1ca50b4618063b1e25548/User_List.png)

This will allow the same functionality to update the user that we already have, but also add the ability for the user to select which reports they will have access too. 

![Edit](/uploads/b844a4838c612de24bf5f7318866f92d/Edit.png)

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 30, 2019, 06:25

@kdjstudios 


Please let us know when we need to post an estimate.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 30, 2019, 11:51

@amishra @smasih @pchaudhary @tim.traylor 

Of course, I will update the labels accordingly. Please focus on the tickets with labels already needing estimates, development, and testing.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jul 25, 2019, 10:02

Waiting on tomorrow's reprioritization meeting before moving forward.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 19, 2019, 23:05

@kdjstudios @jredfield 

Below is the estimate to develop this functionality as per your current mockup provided above.

## Developer Estimate
|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|Medium-1|4|Worked with Concepts, implementation ok|6|High Impact|4.5|10.5|

##### Estimate Notes:
* Risk areas identified: `Report`

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 19, 2019, 23:17

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Nov 20, 2019, 05:41

@kdjstudios @amishra @smasih Before approving this, I just want to confirm with @kdjstudios that the client approved the design per his comment that we would require that approval first.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 20, 2019, 13:59

I will want to review the mock ups and ensure that there is no 'client' info, We can then schedule a call with Jo Ann, to further review the mock ups and upon feedback/approval we can then work with David to create a proposal for the development.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 25, 2019, 13:17

Here are the updated screen mockups. 

![Edit_User](/uploads/b582f1746f4a55f76324cebfe87fa662/Edit_User.png)

![User_List](/uploads/858cd548568c1beffa9c1e9a5f11a972/User_List.png)

![User_Navigation_Button](/uploads/d87b3f534341a330e4fa145ea4b6effe/User_Navigation_Button.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 25, 2019, 13:22

I wrote:

> We do apologize for the delay in following up on this. We encountered some initial design and functionality concerns that we had review internally. We do believe we would be able to create the user configurable report access and we have created a few mock ups of how we anticipate it will look. We would like to review these with you to get your feedback to ensure we fully understand your request. Do you have any available time next week to go over and review the mock ups?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 4, 2019, 12:38

I wrote:

> We are following up to confirm if you are still requesting the ability to have report configurable user permissions? If so, do you have any time available next week to review this request?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 20, 2020, 14:59

I Wrote:

> We are following up to confirm if you are still requesting the ability to have report configurable user permissions? If so, do you have any time available next week to review this request?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 19, 2020, 14:45

Closing ticket.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 19, 2020, 14:45

closed
